File: Problems/1570. Dot Product of Two Sparse Vectors.cpp
class SparseVector {
public:
    
    SparseVector(vector<int> &nums) {
        arr = nums;
    }
    
    // Return the dotProduct of two sparse vectors
    int dotProduct(SparseVector& vec) {
        for(size_t i=0; i<arr.size(); i++){
            sumDP = sumDP + arr[i] * vec.arr[i];
        }
        return sumDP;
    }
private:
    vector<int> arr;
    int sumDP = 0;
};

File: Problems (Python)/1570. Dot Product of Two Sparse Vectors.py
class SparseVector:
    def __init__(self, nums: List[int]):
        self.arr = nums
        

    # Return the dotProduct of two sparse vectors
    def dotProduct(self, vec: 'SparseVector') -> int:
        sumDP = 0
        for i in range(len(self.arr)):
            sumDP = sumDP + self.arr[i] * vec.arr[i]
        return sumDP
